src/Component.js

```diff
@@ -31,10 +31,12 @@ export class CopyToClipboard extends React.PureComponent {
 
     const elem = React.Children.only(children);
 
-    const result = copy(text, options);
+    if (text.trim() !== '') {
```

Comment:
This can be done in the app and does not need to be in the lib.

Example:

```jsx
<CopyToClipboard
  onCopy={this.state.value.trim() !== '' ? this.onCopy : undefined}
  text={this.state.value}>
  <button>Copy to clipboard with button</button>
</CopyToClipboard>
```

Reply:
I tried in codepen with safari, but it still pops a window.
this problem was caused by this line:

```
const result = copy(text, options) // when `text` is empty 
```